### PR TITLE
Implement new button for quick menu

### PR DESCRIPTION
**Pull request by codeimpl**
This patch implements new quick menu button extends android Button
class. It has test code at quick_menu_layout to check new button.

Refs: #983

### CODE REVIEW

File: res/drawable/quick_menu_button_image_ripple.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+     This Source Code Form is subject to the terms of the Mozilla Public
+     License, v. 2.0. If a copy of the MPL was not distributed with this
+     file, You can obtain one at https://mozilla.org/MPL/2.0/. -->
+
+<ripple xmlns:android="http://schemas.android.com/apk/res/android"
+    android:color="@android:color/darker_gray">
+    <item android:id="@android:id/mask">
+        <shape android:shape="rectangle">
+            <corners android:radius="20dp" />
+            <solid android:color="#FFFFFFFF" />
```
Comment (jdragonbae) [inline res/drawable/quick_menu_button_image_ripple.xml:12]: Can we use `@android:color/white`?

File: src/org/triple/banana/quick_menu/QuickMenuNewButton.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import android.util.TypedValue;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.AppCompatButton;
+import androidx.core.content.res.ResourcesCompat;
+
+import org.triple.banana.R;
+
+class QuickMenuNewButton extends AppCompatButton {
+    public QuickMenuNewButton(Context context) {
```
Comment (jdragonbae) [inline src/org/triple/banana/quick_menu/QuickMenuNewButton.java:21]: With annotation @NonNull? `@NonNull Context context`

Comment (codeimpl) [reply src/org/triple/banana/quick_menu/QuickMenuNewButton.java]: done!

File: res/drawable/quick_menu_button_image_ripple.xml (hunk 1 above)
Comment (codeimpl) [reply res/drawable/quick_menu_button_image_ripple.xml]: done!

File: res/layout/quick_menu_layout.xml [hunk 3]
```diff
@@ -25,6 +25,14 @@
             android:layout_height="wrap_content"
             android:layout_weight="1"
             android:gravity="start" />
+
+        <!-- It is test code for QuickMenuNewButton -->
```
Comment (romandev) [inline res/layout/quick_menu_layout.xml:29]: Let's remove this code in this PR if this is working well.

File: src/org/triple/banana/quick_menu/QuickMenuNewButton.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                R.drawable.quick_menu_button_image_background, getContext().getTheme());
+        Drawable[] drawables = {drawable, top};
+        LayerDrawable layerDrawable = new LayerDrawable(drawables);
+        layerDrawable.setLayerInset(1,
+                getResources().getDimensionPixelSize(R.dimen.quick_menu_button_image_padding),
+                getResources().getDimensionPixelSize(R.dimen.quick_menu_button_image_padding),
+                getResources().getDimensionPixelSize(R.dimen.quick_menu_button_image_padding),
+                getResources().getDimensionPixelSize(R.dimen.quick_menu_button_image_padding));
+
+        super.setCompoundDrawablesWithIntrinsicBounds(left, layerDrawable, right, bottom);
```
Comment (romandev) [inline src/org/triple/banana/quick_menu/QuickMenuNewButton.java:54]:
```
Drawable background = ResourcesCompat.getDrawable(getResources(),
                R.drawable.quick_menu_button_image_background, getContext().getTheme());
LayerDrawable iconDrawable = new LayerDrawable(new Drawable[] { background, top});
final int paddingSize = getResources().getDimensionPixelSize(R.dimen.quick_menu_button_image_padding);
iconDrawable.setLayerInset(1, paddingSize, paddingSize, paddingSize, paddingSize);
super.setCompoundDrawablesWithIntrinsicBounds(left, iconDrawable, right, bottom);
```

File: res/values/dimens.xml [hunk 5]
```diff
@@ -20,7 +20,10 @@
     <dimen name="media_remote_top_margin">10dp</dimen>
     <dimen name="media_remote_vertical_seekbar_size">200dp</dimen>
     <dimen name="media_remote_wait_progress_size">96dp</dimen>
+    <dimen name="quick_menu_button_between_image_and_text_padding">10dp</dimen>
     <dimen name="quick_menu_button_image_length">36dp</dimen>
+    <dimen name="quick_menu_button_image_padding">18dp</dimen>
```
Comment (romandev) [inline res/values/dimens.xml:25]: quick_menu_button_background_padding

File: res/values/dimens.xml [hunk 6]
```diff
@@ -20,7 +20,10 @@
     <dimen name="media_remote_top_margin">10dp</dimen>
     <dimen name="media_remote_vertical_seekbar_size">200dp</dimen>
     <dimen name="media_remote_wait_progress_size">96dp</dimen>
+    <dimen name="quick_menu_button_between_image_and_text_padding">10dp</dimen>
```
Comment (romandev) [inline res/values/dimens.xml:23]: quick_menu_button_icon_padding

File: res/drawable/quick_menu_button_image_background.xml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2020 The Triple Banana Authors. All rights reserved.
```
Comment (romandev) [inline res/drawable/quick_menu_button_image_background.xml:2]: 2021

File: res/drawable/quick_menu_button_image_ripple.xml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2020 The Triple Banana Authors. All rights reserved.
```
Comment (romandev) [inline res/drawable/quick_menu_button_image_ripple.xml:2]: 2021

File: src/org/triple/banana/quick_menu/QuickMenuNewButton.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 The Triple Banana Authors. All rights reserved.
```
Comment (romandev) [inline src/org/triple/banana/quick_menu/QuickMenuNewButton.java:1]: 2021

**Review by romandev — APPROVED** [review]
lgtm

File: src/org/triple/banana/quick_menu/QuickMenuNewButton.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+import android.util.TypedValue;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.AppCompatButton;
+import androidx.core.content.res.ResourcesCompat;
+
+import org.triple.banana.R;
+
+class QuickMenuNewButton extends AppCompatButton {
```
Comment (romandev) [inline src/org/triple/banana/quick_menu/QuickMenuNewButton.java:21]: Shouldn't we make `public` for using it in `.xml`?

**Pull request merged**